apis/pkg/v1alpha1/interfaces.go

```diff
@@ -0,0 +1,311 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+	"github.com/crossplane/crossplane/apis/pkg/v1"
+)
+
+// GetCondition of this Function.
+func (f *Function) GetCondition(ct xpv1.ConditionType) xpv1.Condition {
+	return f.Status.GetCondition(ct)
+}
+
+// SetConditions of this Function.
+func (f *Function) SetConditions(c ...xpv1.Condition) {
+	f.Status.SetConditions(c...)
+}
+
+// GetSource of this Function.
+func (f *Function) GetSource() string {
+	return f.Spec.Package
+}
+
+// SetSource of this Function.
+func (f *Function) SetSource(s string) {
+	f.Spec.Package = s
+}
+
+// GetActivationPolicy of this Function.
+func (f *Function) GetActivationPolicy() *v1.RevisionActivationPolicy {
+	return f.Spec.RevisionActivationPolicy
+}
+
+// SetActivationPolicy of this Function.
+func (f *Function) SetActivationPolicy(a *v1.RevisionActivationPolicy) {
+	f.Spec.RevisionActivationPolicy = a
+}
+
+// GetPackagePullSecrets of this Function.
+func (f *Function) GetPackagePullSecrets() []corev1.LocalObjectReference {
+	return f.Spec.PackagePullSecrets
+}
+
+// SetPackagePullSecrets of this Function.
+func (f *Function) SetPackagePullSecrets(s []corev1.LocalObjectReference) {
+	f.Spec.PackagePullSecrets = s
+}
+
+// GetPackagePullPolicy of this Function.
+func (f *Function) GetPackagePullPolicy() *corev1.PullPolicy {
+	return f.Spec.PackagePullPolicy
+}
+
+// SetPackagePullPolicy of this Function.
+func (f *Function) SetPackagePullPolicy(i *corev1.PullPolicy) {
+	f.Spec.PackagePullPolicy = i
+}
+
+// GetRevisionHistoryLimit of this Function.
+func (f *Function) GetRevisionHistoryLimit() *int64 {
+	return f.Spec.RevisionHistoryLimit
+}
+
+// SetRevisionHistoryLimit of this Function.
+func (f *Function) SetRevisionHistoryLimit(l *int64) {
+	f.Spec.RevisionHistoryLimit = l
+}
+
+// GetIgnoreCrossplaneConstraints of this Function.
+func (f *Function) GetIgnoreCrossplaneConstraints() *bool {
+	return f.Spec.IgnoreCrossplaneConstraints
+}
+
+// SetIgnoreCrossplaneConstraints of this Function.
+func (f *Function) SetIgnoreCrossplaneConstraints(b *bool) {
+	f.Spec.IgnoreCrossplaneConstraints = b
+}
+
+// GetControllerConfigRef of this Function.
+func (f *Function) GetControllerConfigRef() *v1.ControllerConfigReference {
+	return nil
+}
+
+// SetControllerConfigRef of this Function.
+func (f *Function) SetControllerConfigRef(*v1.ControllerConfigReference) {}
```

Comment:
I don't think Functions should support ControllerConfigs. We know we don't want to pursue them, and we have [a design for their replacement](https://github.com/crossplane/crossplane/blob/51016ddd75c7a03f56e5da27edf55bab7847a9ee/design/one-pager-package-runtime-config.md).

I'm hoping we can build the runtime config implementation as essentially part of the Functions work for v1beta1. If not though, I'd prefer v1beta1 Functions to be pretty static controller config wise over propagating use of the ControllerConfig type.

Reply:
If we feel pretty confident that we can implement the runtime config design in time for v1.14 I'd feel okay merging this as-is and refactoring before v1.14.